### PR TITLE
Add new default feature preset and updates for new `matminer` & `pymatgen` versions

### LINKED ISSUE #102: CI tests are very slow
We should check that parallelism in the tests isn't thrashing GitHub actions, and also that the test cases we have are suitable. The timing of each test is produced in the CI logs (currently dominated by the GA and bootstrap tests).

### PR DESCRIPTION
**Pull request by ml-evs**
This PR updates most of the core dependencies of MODNet to the latest versions, plus a few associated changes:

- Disable slow tests in CI and uses a pip cache to accelerate install time (closes #102)
- Add a new `Matminer2023Featurizer`/`CompositionOnlyMatminer2023Featurizer` that mimics the old `DeBreuck2020Featurizer` but uses the latest matminer featurizer definitions (same featurizers, but potentially different values and different representations required).
- Makes `Matminer2023` the new default preset.
- Adds a warning when loading `DeBreuck2020Featurizer` with an incompatible matminer version (i.e. >0.6).

### CODE REVIEW

**Comment by ml-evs** [comment]
This is currently held up by https://github.com/hackingmaterials/matminer/pull/896 and https://github.com/hackingmaterials/matminer/issues/864, but I've done the appropriate plumbing, defined a new featurizer for 2023 and updated the tests.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,17 +11,22 @@ jobs:
...
+        cache: 'pip'
+        cache-dependency-path: |
+          **/requirements*.txt
 
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install -r requirements.txt
-        pip install --ignore-installed .[test,dev]
+        pip install .[test,dev]
```
Comment (ppdebreuck) [inline .github/workflows/ci.yml:29]: --ignore-installed not required anymore ?

File: README.md [hunk 2]
```diff
@@ -47,10 +47,10 @@ Activate the environment:
...
 
-Then, install pymatgen v2020.8.13 with conda, which will bundle several pre-built dependencies (e.g., numpy, scipy):
+Then, install pymatgen from conda-forge, which will bundle several pre-built dependencies (e.g., numpy, scipy):
 
 ```shell
-conda install -c conda-forge pymatgen=2020.8.13
+conda install -c conda-forge pymatgen
 ```
 
 (you could alternatively do this step with `pip install pymatgen==2020.8.13`).
```
Comment (ppdebreuck) [inline README.md:56]: This line can probably be removed or just `pip install pymatgen`

File: modnet/featurizers/presets/debreuck_2020.py [hunk 3]
```diff
@@ -1,26 +1,57 @@
...
+
+        """
+        import matminer
+
+        if matminer.__version__ != self.package_version_requirements["matminer"]:
+            warnings.warn(
+                f"The {self.__class__.__name__} preset was written for and tested only with matminer{self.package_version_requirements['matminer']}.\n"
+                "Newer versions of matminer will not work, and older versions may not be compatible with newer MODNet versions due to other conflicts.\n"
+                "To use this featurizer robustly, please install `modnet==0.1.13` with its pinned dependencies.\n\n"
+                "This preset will now be initialised without importing matminer featurisers to enable use with existing previously featurized data, "
```
Comment (ppdebreuck) [inline modnet/featurizers/presets/debreuck_2020.py:43]: "featurizers"

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ml-evs) [reply .github/workflows/ci.yml:29]: No, this was actually the cause of a bug... I interpreted "--ignore-installed" to mean, "do not install things again if they exist with another version", but actually it means "install over things even if they already exist" :sweat_smile:

File: README.md (hunk 2 above)
Comment (ml-evs) [reply README.md:56]: I'll remove it, the big afterwards it also out of date now

File: modnet/featurizers/presets/debreuck_2020.py (hunk 3 above)
Comment (ml-evs) [reply modnet/featurizers/presets/debreuck_2020.py:43]: Are you commenting on my British spelling of featurisers or my lack of plural in line 42?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ppdebreuck) [reply .github/workflows/ci.yml:29]: I was also confused by the interpretation :p

File: modnet/featurizers/presets/debreuck_2020.py (hunk 3 above)
Comment (ppdebreuck) [reply modnet/featurizers/presets/debreuck_2020.py:43]: US vs British spelling 😅

Comment (ppdebreuck) [reply modnet/featurizers/presets/debreuck_2020.py:43]: Line 43 sorry

File: modnet/featurizers/presets/debreuck_2020.py [hunk 4]
```diff
@@ -1,26 +1,57 @@
...
+                "but attempts to perform further featurization will result in an error."
+            )
 
+        else:
+            super().__init__()
+            self.load_featurizers()
+            self.fast_oxid = fast_oxid
+
+    def load_featurizers(self):
         with contextlib.redirect_stdout(None):
```
Comment (ppdebreuck) [inline modnet/featurizers/presets/debreuck_2020.py:53]: Great solution to unnecessary printing :)

**Review by ppdebreuck — APPROVED** [review]
Thanks @ml-evs !

**Pull request merged**